### ISSUE TITLE
OSError: Invalid data stream when updating

### ISSUE THREAD

**Issue by manuvazquez** [conda/conda#1625]
Hi,

when I tried to update everything in my conda installation by

conda update --all

I get an error while "Extracting packages". Below is the traceback and also my "conda info"

Traceback (most recent call last):
  File "/export/usuarios01/mvazquez/anaconda3/bin/conda", line 5, in <module>
    sys.exit(main())
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/cli/main.py", line 194, in main
    args_func(args, p)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/cli/main.py", line 201, in args_func
    args.func(args, p)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/cli/main_update.py", line 38, in execute
    install.install(args, parser, 'update')
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/cli/install.py", line 417, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/plan.py", line 533, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/instructions.py", line 148, in execute_instructions
    cmd(state, arg)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/instructions.py", line 62, in EXTRACT_CMD
    install.extract(config.pkgs_dirs[0], arg)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/site-packages/conda/install.py", line 456, in extract
    t.extractall(path=path)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/tarfile.py", line 1980, in extractall
    self.extract(tarinfo, path, set_attrs=not tarinfo.isdir())
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/tarfile.py", line 2019, in extract
    set_attrs=set_attrs)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/tarfile.py", line 2088, in _extract_member
    self.makefile(tarinfo, targetpath)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/tarfile.py", line 2134, in makefile
    copyfileobj(source, target, tarinfo.size)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/tarfile.py", line 239, in copyfileobj
    buf = src.read(BUFSIZE)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/bz2.py", line 293, in read
    return self._read_block(size)
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/bz2.py", line 254, in _read_block
    while n > 0 and self._fill_buffer():
  File "/export/usuarios01/mvazquez/anaconda3/lib/python3.4/bz2.py", line 218, in _fill_buffer
    self._buffer = self._decompressor.decompress(rawblock)
OSError: Invalid data stream

Current conda install:

             platform : linux-64
        conda version : 3.17.0
  conda-build version : 1.13.0
       python version : 3.4.3.final.0
     requests version : 2.7.0
     root environment : /export/usuarios01/mvazquez/anaconda3  (writable)
  default environment : /export/usuarios01/mvazquez/anaconda3
     envs directories : /export/usuarios01/mvazquez/anaconda3/envs
        package cache : /export/usuarios01/mvazquez/anaconda3/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

Any help is appreciated...

Cheers!!

**Comment by asmeurer** [comment]
Did it say what package it was extracting when it crashed?

**Comment by manuvazquez** [comment]
Hi Aaron,

...and thanks for your prompt reply!!

I just tried again and it crashed at the very beginning (0%)... However, I just found out that some of the .bz2 in anaconda/pkgs are corrupted :-/ I had thought of this before and renamed pkgs to bak.pkgs to force anaconda to download everything again but it did not worked, and the reason is that in every download some of the files were corrupted. So, this looks like a hardware issue, and not related at all with anaconda....sorry for the noise. 

Cheers.

PS: maybe it would be a good thing that conda used some kind of checksum on the downloaded files...although this is probably no big deal (no rant intended :) ).
